### PR TITLE
Graph RBAC Cont'd

### PR DESCRIPTION
**Pull request by jianghaolu**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/ResourceManagement/Graph.RBAC/ServicePrincipalImpl.cs [hunk 1]
```diff
@@ -80,9 +95,28 @@ public ServicePrincipalImpl WithNewRoleInSubscription(BuiltInRole role, string s
...
+            if (cachedPasswordCredentials.ContainsKey(name))
+            {
+                passwordCredentialsToDelete.Add(name);
+            }
+            else if (cachedCertificateCredentials.ContainsKey(name))
+            {
+                certificateCredentialsToDelete.Add(name);
+            }
+            return this;
         }
```
Comment (hovsepm) [inline src/ResourceManagement/Graph.RBAC/ServicePrincipalImpl.cs]: I lost my vision :) Please format document. in VS select all then Ctrl+K,D

File: src/ResourceManagement/Graph.RBAC/Generated/DomainsOperationsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            /// </summary>
+            /// <param name='operations'>
+            /// The operations group for this extension method.
+            /// </param>
+            /// <param name='odataQuery'>
+            /// OData parameters to apply to the operation.
+            /// </param>
+            public static IEnumerable<DomainInner> List(this IDomainsOperations operations, ODataQuery<UserInner> odataQuery = default(ODataQuery<UserInner>))
+            {
+                return ((IDomainsOperations)operations).ListAsync(odataQuery).GetAwaiter().GetResult();
```
Comment (hovsepm) [inline src/ResourceManagement/Graph.RBAC/Generated/DomainsOperationsExtensions.cs]: >urn ((IDomainsOperations)operations).ListAsync(odataQuery).GetAwaiter().GetResult(); [](start = 19, length = 84)

please remove all generated Sync Methods

**Review by hovsepm — CHANGES_REQUESTED** [review]
please remove all generated Sync Methods

File: src/ResourceManagement/Graph.RBAC/Generated/DomainsOperationsExtensions.cs (hunk 2 above)
Comment (jianghaolu) [inline src/ResourceManagement/Graph.RBAC/Generated/DomainsOperationsExtensions.cs]: Is there a version of AutoRest I should use?

**Comment by jianghaolu** [comment]
@hovsepm addressed. Please review, thanks!

**Review by hovsepm — APPROVED** [review]

**Pull request merged**